titus-supplementary-component/tasks-publisher/src/main/java/com/netflix/titus/supplementary/taskspublisher/EsPublisher.java

```diff
@@ -29,86 +27,56 @@
 import com.netflix.spectator.api.Functions;
 import com.netflix.spectator.api.Registry;
 import com.netflix.spectator.api.patterns.PolledMeter;
-import com.netflix.titus.api.jobmanager.JobAttributes;
 import com.netflix.titus.common.util.rx.ReactorExt;
 import com.netflix.titus.common.util.rx.RetryHandlerBuilder;
-import com.netflix.titus.common.util.tuple.Pair;
 import com.netflix.titus.ext.elasticsearch.TaskDocument;
-import com.netflix.titus.grpc.protogen.Job;
-import com.netflix.titus.grpc.protogen.Task;
-import com.netflix.titus.runtime.endpoint.v3.grpc.V3GrpcModelConverters;
 import org.eclipse.jetty.util.ConcurrentHashSet;
 import org.reactivestreams.Publisher;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Qualifier;
 import reactor.core.Disposable;
+import reactor.core.publisher.ConnectableFlux;
 import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
 import reactor.core.scheduler.Schedulers;
 
-
-public class TasksPublisherCtrl {
-    private static final Logger logger = LoggerFactory.getLogger(TasksPublisherCtrl.class);
+public class EsPublisher implements TasksPublisher {
```

Comment:
Singleton?

Reply:
Dropping an explicit declaration since spring default scope is Singleton.